var/spack/repos/builtin/packages/heimdall/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Heimdall(AutotoolsPackage, CudaPackage):
+    """GPU accelerated transient detection pipeline"""
+
+    homepage = "https://sourceforge.net/projects/heimdall-astro/"
+    git = "https://git.code.sf.net/p/heimdall-astro/code"
+
+    version("1.0", branch="master")
+
+    conflicts("~cuda", msg="You must specify +cuda")
+    conflicts("cuda_arch=none", when="+cuda", msg="You must specify the CUDA architecture")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("cuda@:11.7")
+
+    # Pass the cuda architecture to DEDISP and PSRDADA for building
+    for arch in CudaPackage.cuda_arch_values:
+        depends_on("dedisp +cuda cuda_arch=%s" % arch,
+                   when="+cuda cuda_arch=%s" % arch)
+
+        depends_on("psrdada +cuda cuda_arch=%s" % arch,
+                   when="+cuda cuda_arch=%s" % arch)
```

Comment:
We prefer f-strings now.
```suggestion
        depends_on(f"dedisp +cuda cuda_arch={arch}",
                   when=f"+cuda cuda_arch={arch}")

        depends_on(f"psrdada +cuda cuda_arch={arch}",
                   when=f"+cuda cuda_arch={arch}")
```